Cesium3DTilesContent/src/I3dmToGltfConverter.cpp

```diff
@@ -0,0 +1,738 @@
+    const CesiumGltfReader::GltfReaderOptions& options,
+    const ConverterSubprocessor& subprocessor,
+    GltfConverterResult& result) {
+  ConvertResult subResult;
+  DecodedInstances& decodedInstances = subResult.decodedInstances;
+  subResult.gltfResult = result;
+  auto finishEarly = [&]() {
+    return subprocessor.asyncSystem.createResolvedFuture(std::move(subResult));
+  };
+  if (header.featureTableJsonByteLength == 0 ||
+      header.featureTableBinaryByteLength == 0) {
+    return finishEarly();
+  }
+  const uint32_t glTFStart = headerLength + header.featureTableJsonByteLength +
+                             header.featureTableBinaryByteLength +
+                             header.batchTableJsonByteLength +
+                             header.batchTableBinaryByteLength;
+  const uint32_t glTFEnd = header.byteLength;
+  auto gltfData = instancesBinary.subspan(glTFStart, glTFEnd - glTFStart);
+  std::optional<CesiumAsync::Future<ByteResult>> assetFuture;
+  auto featureTableJsonData =
+      instancesBinary.subspan(headerLength, header.featureTableJsonByteLength);
+  rapidjson::Document featureTableJson;
+  featureTableJson.Parse(
+      reinterpret_cast<const char*>(featureTableJsonData.data()),
+      featureTableJsonData.size());
+  if (featureTableJson.HasParseError()) {
+    subResult.gltfResult.errors.emplaceError(fmt::format(
+        "Error when parsing feature table JSON, error code {} at byte offset "
+        "{}",
+        featureTableJson.GetParseError(),
+        featureTableJson.GetErrorOffset()));
+    return finishEarly();
+  }
+  InstanceContent parsedContent;
+  // Global semantics
+  if (auto optinstancesLength =
+          getValue<uint32_t>(featureTableJson, "INSTANCES_LENGTH")) {
+    parsedContent.instancesLength = *optinstancesLength;
+  } else {
+    subResult.gltfResult.errors.emplaceError(
+        "Error parsing I3DM feature table, no valid "
+        "INSTANCES_LENGTH was found.");
+    return finishEarly();
+  }
+  parsedContent.rtcCenter =
+      parseArrayValueDVec3(featureTableJson, "RTC_CENTER");
+  decodedInstances.rtcCenter = parsedContent.rtcCenter;
+
+  parsedContent.position =
+      parseOffset(featureTableJson, "POSITION", subResult.gltfResult.errors);
+  if (!parsedContent.position) {
+    if (subResult.gltfResult.errors.hasErrors()) {
+      return finishEarly();
+    }
+    parsedContent.positionQuantized = parseOffset(
+        featureTableJson,
+        "POSITION_QUANTIZED",
+        subResult.gltfResult.errors);
+    if (subResult.gltfResult.errors.hasErrors()) {
+      return finishEarly();
+    }
+  }
+  if (parsedContent.positionQuantized) {
+    parsedContent.quantizedVolumeOffset =
+        parseArrayValueDVec3(featureTableJson, "QUANTIZED_VOLUME_OFFSET");
+    if (!parsedContent.quantizedVolumeOffset) {
+      subResult.gltfResult.errors.emplaceError(
+          "Error parsing I3DM feature table, No valid "
+          "QUANTIZED_VOLUME_OFFSET property");
```

Comment:
Just a bit more explicit 

```suggestion
          "Error parsing I3DM feature table, the I3dm uses quantized positions "
          "but has no valid QUANTIZED_VOLUME_OFFSET property");
```